dbos/_dbos_config.py

```diff
@@ -33,11 +116,30 @@ class DatabaseConfig(TypedDict, total=False):
     ssl: Optional[bool]
     ssl_ca: Optional[str]
     local_suffix: Optional[bool]
-    app_db_client: Optional[str]
     migrate: Optional[List[str]]
     rollback: Optional[List[str]]
 
 
+def parse_db_string_to_dbconfig(db_string: str) -> DatabaseConfig:
+    db_url = make_url(db_string)
+    db_config = {
+        "hostname": db_url.host,
+        "port": db_url.port or 5432,
+        "username": db_url.username,
+        "password": db_url.password,
+        "app_db_name": db_url.database,
+    }
+    for key, value in db_url.query.items():
+        str_value = value[0] if isinstance(value, tuple) else value
+        if key == "connect_timeout":
+            db_config["connectionTimeoutMillis"] = int(str_value) * 1000
+        elif key == "sslmode":
+            db_config["ssl"] = str_value == "require"
+        elif key == "sslcert":
+            db_config["ssl_ca"] = str_value
+    return cast(DatabaseConfig, db_config)
```

Comment:
In the future we can extend this to support arbitrary connection configuration properties.